magento/pwa-studio
# [bug]: Fix broken summarizeEvents function

## Description

A bug in the `summarizeEvents` function was causing the dev server to exit when switching branches and/or running `yarn install`. This occurred because the reducer function had malformed arguments and also because the reducer was incorrectly not returning the accumulator value after modifying it.

## Related Issue





Closes [PWA-405](https://jira.corp.magento.com/browse/PWA-405).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


**Prereq**: in `watch-all.js` change line 81 to `eventBuffer.length >= 1 ? summarizeEvents() : eventBuffer;` This will cause the function to run so we can test it.

1. Run `yarn watch:all` on this branch.
2. In a different terminal, switch branches ie `git checkout origin/develop~1`
3. Run `yarn install`
4. At no point should the watcher have exited.

## Screenshots / Screen Captures (if appropriate)
Yay!

[![Image from Gyazo](https://i.gyazo.com/c780265ed3527db47f2ec5cb779aae0f.png)](https://gyazo.com/c780265ed3527db47f2ec5cb779aae0f)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
